### ISSUE TITLE
Modularize git repo support

### ISSUE THREAD

**Issue by harrism** [trxcllnt/rapids-compose#4]
Currently rapids-compose clones a fixed set of repos. Most users are RAPIDS developers, and often they work on a different subset of those repos. So there's a lot of overhead on the initial cloning, but also whenever you re-run make init.

It would be nice to first select which RAPIDS repos you want to work with. It would also be nice to be able to "turn off" repos that are already cloned, perhaps by rerunning make init.

**Comment by trxcllnt** [comment]
Agreed. The static nature of docker-compose.yml makes this non-trivial, but I should be able to work around that with some docker-in-docker magic.

**Comment by jrhemstad** [comment]
Similarly, it would be nice if you opt out of a project the conda environment solver would ignore the conda environment of that project. I frequently run into issues where projects have incompatible conda environments.
